File: app/src/main/java/dev/olog/basil/presentation/main/RecipesViewPagerAdapter.java
package dev.olog.basil.presentation.main;

import java.util.List;

import androidx.databinding.ViewDataBinding;
import dev.olog.basil.BR;
import dev.olog.basil.R;
import dev.olog.basil.presentation.base.BaseAdapter;
import dev.olog.basil.presentation.model.DisplayableRecipeImage;
import dev.olog.basil.presentation.widget.ParallaxScrimImageView;

public class RecipesViewPagerAdapter extends BaseAdapter<DisplayableRecipeImage> {

    public RecipesViewPagerAdapter() {
    }

    public RecipesViewPagerAdapter(List<DisplayableRecipeImage> data) {
        super(data);
    }

    @Override
    protected void initViewHolderListeners(ViewDataBinding binding) {
        ParallaxScrimImageView imageView = binding.getRoot().findViewById(R.id.image);
    }

    @Override
    protected void bind(ViewDataBinding binding, DisplayableRecipeImage item, int position) {
        binding.setVariable(BR.recipe, item);
    }

    @Override
    protected int provideLayoutId() {
        return R.layout.item_recipe_image;
    }
}

File: app/src/main/java/dev/olog/basil/presentation/main/RecipesViewPagerAdapter.kt
package dev.olog.basil.presentation.main

import androidx.databinding.ViewDataBinding
import com.bumptech.glide.Glide
import dev.olog.basil.R
import dev.olog.basil.presentation.base.BaseAdapter
import kotlinx.android.synthetic.main.item_recipe_image.view.*

class RecipesViewPagerAdapter : BaseAdapter<Int>() {

    override fun initViewHolderListeners(binding: ViewDataBinding) {

    }

    override fun bind(binding: ViewDataBinding, item: Int, position: Int) {
        Glide.with(binding.root)
                .load(item)
                .into(binding.root.image)
    }

    override fun provideLayoutId(): Int {
        return R.layout.item_recipe_image
    }
}
